### DISCUSSION TITLE
一些关于小雅代码的问题请教

### DISCUSSION THREAD

**Discussion by AddMoreChili** [AI4Finance-Foundation/ElegantRL#279] [Q&A]
1、在net.py中为什么只对网络最后一层做正交初始化？请问这是基于什么样的考虑？
2、QNetDuel中，Q值计算使用的是value = q_val - q_val.mean(dim=1, keepdim=True) + q_adv  貌似应该是value = q_val - q_adv.mean(dim=1, keepdim=True) + q_adv

**Comment by Yonv1943** [comment]
1. 可以对所有网络层做正交初始化，也可以对其中一层做正交初始化，也可以不做。解释可以看 [影响PPO算法性能的10个关键技巧（附PPO算法简洁Pytorch实现） Trick 8—Orthogonal Initialization 正交初始化](https://zhuanlan.zhihu.com/p/512327050)
2. 你是对的，我把 `q_value` 和 `q_advantage` 写反了，我需要改过来，谢谢提醒
